src/shared/find-esp.c

```diff
@@ -238,32 +238,28 @@ static int verify_esp_udev(
 }
 
 static int verify_fsroot_dir(
+                int dir_fd,
                 const char *path,
                 bool searching,
                 bool unprivileged_mode,
                 dev_t *ret_dev) {
 
-        _cleanup_close_ int fd = -EBADF;
         STRUCT_NEW_STATX_DEFINE(sxa);
         STRUCT_NEW_STATX_DEFINE(sxb);
         int r;
 
         /* Checks if the specified directory is at the root of its file system, and returns device
          * major/minor of the device, if it is. */
 
+        assert(dir_fd >= 0 || dir_fd == AT_FDCWD);
```

Comment:
All functions called in `verify_fsroot_dir()` support AT_FDCWD. But calling it with AT_FDCWD is meaningless. I think it's better to refuse the case.